frontera/contrib/backends/redis_backend/__init__.py

```diff
@@ -129,21 +157,9 @@ def get_next_requests(self, max_n_requests, partition_id, **kwargs):
                 request.meta[FIELD_SCORE] = score
                 results.append(request)
         if len(to_remove) > 0:
-            timeout = _get_retry_timeouts()
-            while True:
-                try:
-                    connection = StrictRedis(connection_pool=self._pool)
-                    connection.zrem(partition_id, *to_remove)
-                    break
-                except ConnectionError as e:
-                    self._logger.exception("Connection to Redis failed when attempting to remove scheduled items")
-                    pause = timeout.next()
-                    if pause == None:
-                        break
-                    sleep(pause)
-                except ResponseError as e:
-                    self._logger.exception("Writing to Redis failed when attempting to remove scheduled items")
-                    break
+            self._redis_operation(
```

Comment:
Overall approach looks much better to me!
However, we could make it more pythonic. See:
```python
class RedisOperation(object):

    def __init__(self, ...):
        self._connection = ...

    def __getattr__(self, _api):
        if _api in self._connection.__dir__():
            return functools.partial(self._redis_operation, _api)

        raise AttributeError("{} instance has no attribute '{}'".format(
            self.__class__.__name__, _api))

    def __dir__(self):
        return self._connection.__dir__()

    def _redis_operation(self, _api, *args, **kwargs):
        return self._connection.get(_api)((*args, **kwargs)

```